frontend/src/components/ResourceInfo.tsx

```diff
@@ -71,7 +71,16 @@ export class ResourceInfo extends React.Component<ResourceInfoProps, {}> {
     const customPropertyMap = resource.getCustomPropertiesMap();
     return (
       <section>
-        <h1 className={commonCss.header}>Type: {this.props.typeName}</h1>
+        <h1 className={commonCss.header}>
+          {this.props.typeName}
+          {(() => {
+            const stateText = getResourceStateText(this.props);
+            if (stateText) {
+              return ` (${stateText})`;
+            }
+            return '';
+          })()}
```

Comment:
Can this part be extracted to a const var? Hope that it makes the element rendering logic more concise:

```
let resourceTitle = this.props.typeName;
const stateText = getResourceStateText(this.props);
if (stateText) {
  resourceTitle = resourceTitle + ` (${stateText})`;
}

return (
      ...
        <h1 className={commonCss.header}>
          {resourceTitle}
        </h1>
      ....
   )
```

Reply:
Good point, will change as you suggested